test/test_generators.py

```diff
@@ -40,6 +41,186 @@ def create_test_image(dirname, array, normalization=None):
     return image
 
 
+def create_mono_image(dirname, array, normalization=None):
+    """Create a test Image instance."""
+    filename = str(dirname / 'tmp.tif')
+    create_test_file(filename, array)
+    satellite = 'monochrome'
+    image = Image(filename, satellite, normalization_parameters=normalization)
+    return image
+
+
+# 'image_shape','step_size','window_shape','window_ref_arrays'
+WINDOW_TEST_DATA = [
```

Comment:
Make this a list of dicts (instead of list of tuples), that would be much more readable:
```python
WINDOW_TEST_DATA = [
   {
      'image_shape': (3, 3),
      'step_size': (2, 1),
       'window_shape': (3, 4),
       'windows': [
         np.ma.array([[0, 0, 0, 1],
                      [0, 0, 3, 4],
                      [0, 0, 6, 7]],
                     mask=[[True, True, False, False],
                           [True, True, False, False],
                           [True, True, False, False]]),
...
    },
...
]
```